rpc-server/src/main.rs

```diff
@@ -19,10 +19,31 @@ mod utils;
 // Categories for logging
 pub(crate) const RPC_SERVER: &str = "read_rpc_server";
 
-macro_rules! handle_rpc_method {
-    ($method:expr, $method_name:expr) => {
-        |data, params| async move { handle_rpc_method($method, $method_name, data, params).await }
-    };
+trait WithMethodAndMetrics {
+    fn with_method_and_metrics<F, Fut, T, P>(self, method_name: &'static str, method: F) -> Self
+    where
+        F: Fn(Data<config::ServerContext>, Params<P>) -> Fut + Clone + Send + Sync + 'static,
+        Fut: std::future::Future<Output = Result<T, crate::errors::RPCError>> + Send + 'static,
+        T: serde::Serialize + Send + 'static,
+        P: serde::de::DeserializeOwned + Send + 'static;
+}
+
+impl<R> WithMethodAndMetrics for jsonrpc_v2::ServerBuilder<R>
+where
+    R: Router,
+{
+    fn with_method_and_metrics<F, Fut, T, P>(self, method_name: &'static str, method: F) -> Self
+    where
+        F: Fn(Data<config::ServerContext>, Params<P>) -> Fut + Clone + Send + Sync + 'static,
+        Fut: std::future::Future<Output = Result<T, crate::errors::RPCError>> + Send + 'static,
+        T: serde::Serialize + Send + 'static,
+        P: serde::de::DeserializeOwned + Send + 'static,
+    {
+        self.with_method(method_name, move |data, params| {
+            let method = method.clone();
```

Comment:
Ideally, I want to remove this `.clone()`, but it doesn't work without this workaround

Reply:
Probably new approach with `&'static F` is even worse, so I'll consider using `Rc`/`Arc` instead